### ISSUE TITLE
No warning on checking unintroduced variables

### ISSUE THREAD

**Issue by adrianleh** [egraphs-good/egglog#346]
**Current behavior**

When trying to check two variables that have not been introduced the egraph manually the check quietly fails.

Example:
```
(datatype Dim (Plus Dim Dim) (NamedDim String) (Lit i64))
(run 5)
(check (= (Lit 1) (Lit 1)))
```

This can be fixed by explicitly adding the node to the egraph 
```
(datatype Dim (Plus Dim Dim) (NamedDim String) (Lit i64))
(run 5)
(Lit 1)
(check (= (Lit 1) (Lit 1)))
```
(Thanks @saulshanabrook)

**Expected behavior**
It would be great if either the nodes could be implicitly introduced to the egraph or if there would be at least a warning informing the user about this potential issue

(from https://egraphs.zulipchat.com/#narrow/stream/375765-egg.2Fegglog/topic/First.20time.20user.20of.20python.20bindings/near/421896795)

**Comment by oflatt** [comment]
Was this fixed by #344?
I think it was, based on testing this in the web demo.

**Comment by saulshanabrook** [comment]
I was wondering the same thing... but it seems like it still is not working? Unless I am just getting an outdated version somehow:

![Screenshot 2024-02-16 at 11 54 14 AM](https://github.com/egraphs-good/egglog/assets/1186124/12038dac-f169-40e8-9e77-d862dd723c93)

**Comment by adrianleh** [comment]
I was also getting this issue online and locally with the version I built from main today.

**Comment by oflatt** [comment]
Oh that's the behavior I expected.
`check` runs a query, so if the two terms are not in the database it fails

**Comment by oflatt** [comment]
Perhaps we could improve the documentation or issue a warning, as suggested?
When / how would we detect when a warning should be issued?

**Comment by adrianleh** [comment]
I am not familiar with the inner working off egglog so take this with a grain of salt but naïvely I would think a warning should be issued when running check if there is a term within the checked expressions that is not in the database

**Comment by oflatt** [comment]
That's a good idea- we could detect when there's a term with no pattern variables in it.

**Comment by yihozhang** [comment]
Yeah, the current behavior is useful when you want to check whether a certain tuple is in the database (e.g., a tuple that would be added when something is violated).

**Comment by adrianleh** [comment]
That makes sense, however, I think a warning would still make sense especially for new users. Advanced users who desire this behavior can then ignore/suppress this warning
